### PR TITLE
Simplify task waiting procedure

### PR DESCRIPTION
**Pull request by mdellweg**
Calculates task deadline upfront.

[noissue]

Note: This is to understand a nightly failure on the 3.18 better.

Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

File: pulp-glue/pulp_glue/common/context.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def __init__(
         api_root: str,
         api_kwargs: Dict[str, Any],
         background_tasks: bool,
-        timeout: int,
+        timeout: Union[int, datetime.timedelta],
```
Comment (lubosmj) [inline pulp-glue/pulp_glue/common/context.py:123]: I like the change overall. But, can you provide an example where `datetime.timedelta` will be used?

Comment (mdellweg) [reply pulp-glue/pulp_glue/common/context.py:123]: Well it's an extension to the current interface, currently used nowhere. It felt like a low hanging fruit to me.

**Review by lubosmj — APPROVED** [review]
This PR makes the code cleaner. Still, I would leave the fruit hanging low without the need for `datetime.timedelta`. 🙈

**Comment by mdellweg** [comment]
> This PR makes the code cleaner. Still, I would leave the fruit hanging low without the need for `datetime.timedelta`. 🙈

I would like to keep it, because having the same variable name being internally represented as timedelta, but not accepting a timedelta from an external call seems not nice. OTOH, we need to keep the interface accepting int, because that is what we can get from the config (and also the interface of the glue library we need to maintain).

**Review by ggainey — APPROVED** [review]

**Pull request merged**